File: haskell/ITP1_10_B.hs
ans :: [Double] -> [Double]
ans (a:b:ac:_) =
  let c = sqrt (a*a + b*b - 2*a*b*cos(2*pi*ac/360.0))
      s = 0.5 * a * b * sin(2*pi*ac/360.0)
      l = a + b + c
      h = 2 * s / a
  in
    (s:l:h:[])

main = do
  l <- getLine
  let i = map read $ words l :: [Double]
      o = ans i
  mapM_ print o

File: python/ITP1_10_B.py
import math

[a,b,c] = list(map(float,input().split()))

r = c / 360.0 * 2 * math.pi
s = a * b * math.sin(r) / 2
h = s / a * 2

i = math.sqrt( h*h + (a - b*math.cos(r))*(a - b*math.cos(r)))
l = a + b + i

print(s)
print(l)
print(h)
